ddl/index.go

```diff
@@ -599,6 +619,19 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 	return ver, errors.Trace(err)
 }
 
+func doReorgWorkForCreateIndexMultiSchema(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job,
+	tbl table.Table, indexInfo *model.IndexInfo) (done bool, ver int64, err error) {
+	if job.MultiSchemaInfo.Revertible {
+		done, ver, err = doReorgWorkForCreateIndex(w, d, t, job, tbl, indexInfo)
+		if done {
+			job.MarkNonRevertible()
+			done = false // We need another round to wait for all the others sub-jobs to finish.
+		}
+		return done, ver, err
```

Comment:
Could we do as following change?

```suggestion
		if done {
			job.MarkNonRevertible()
		}
		// We need another round to wait for all the others sub-jobs to finish.
		return false, ver, err
```